### PR TITLE
refactor set_opts_cwd function so that git is called only once if cwd is a git dir

### PR DESCRIPTION
**Pull request by smartding**
The original implementation of set_opts_cwd calls git command twice if current dir is a git dir:
1. `git rev-parse HEAD` : to determine if current dir is a git dir
2. `git rev-parse --show-toplevel`: to find root of git dir

Can we remove the first call and just call `git rev-parse --show-toplevel` instead? It  returns an error message if not we are not in a git dir anyway.

### CODE REVIEW

**Comment by Conni2461** [comment]
I see what you mean, but we still have to do a check if `opt.cwd` is set.

File: lua/telescope/builtin/git.lua [hunk 1]
```diff
@@ -144,20 +144,18 @@ git.status = function(opts)
...
-    opts.cwd = vim.fn.systemlist("git rev-parse --show-toplevel")[1]
+    opts.cwd = vim.fn.expand('%:p:h')
+  end
+
+  -- Find root of git directory and remove trailing newline characters
+  opts.cwd = vim.fn.systemlist("git -C " .. opts.cwd .. " rev-parse --show-toplevel")[1]
+
+  -- opts.cwd contains the error message from git command if not in a git dir
+  if 1 ~= vim.fn.isdirectory(opts.cwd) then
+    error(opts.cwd)
```
Comment (Conni2461) [inline lua/telescope/builtin/git.lua:158]: This results in a "not" good error message. For me
```
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Comment (Conni2461) [reply lua/telescope/builtin/git.lua:158]: And checking for `vim.v.shell_error` is way better than isdirectory.
Could you keep the path around so we can get `error(path .. ' is not a git directory')` and if ret val is 0 than set opts.cwd?
Do you know what i mean

Comment (smartding) [reply lua/telescope/builtin/git.lua]: ok, the new commit uses `vim.v.shell_error` now, I also kept path around so that the error message is better.

**Comment by Conni2461** [comment]
Thanks for that. Works expected.
Also thanks for basically reviewing my code :)

**Pull request merged**